api/types/system_role.go

```diff
@@ -61,6 +60,27 @@ const (
 	RoleWindowsDesktop SystemRole = "WindowsDesktop"
 )
 
+// roleMappings maps a set of allowed lowercase system role names
+// to the proper system role
+var roleMappings = map[string]SystemRole{
+	"auth":            RoleAuth,
+	"node":            RoleNode,
+	"proxy":           RoleProxy,
+	"admin":           RoleAdmin,
+	"provisiontoken":  RoleProvisionToken,
```

Comment:
No, it never has been.

We're a little more tolerant and allow `_` separators for things that are commonly typed on the command line, like when generating tokens to allow a kube service or windows desktop service to join the cluster.

Since no one ever needs to generate provision tokens I didn't see a need to make `provision_token` valid.

In fact, after a quick search, it seems `RoleProvisionToken` is never used. Maybe we should delete it.